src/modules/Dashboard/features/Participant/Assignments/ByParticipant/ByParticipant.tsx

```diff
@@ -1,5 +1,181 @@
+  const { result: respondentSubject } = useSubject() ?? {};
+  const [expandedViewsData, setExpandedViewsData] = useState<
+    Record<string, TargetSubjectsByRespondent | undefined>
+  >({});
+  const [expandedViewsLoading, setExpandedViewsLoading] = useState<Record<string, boolean>>({});
+
+  const {
+    execute: fetchActivities,
+    isLoading: isLoadingActivities,
+    value: fetchedActivities,
+  } = useAsync(getAppletRespondentSubjectActivitiesApi, { retainValue: true });
+
+  const activities = fetchedActivities?.data.result ?? [];
+
+  const handleRefetchExpandedView = useCallback(
+    async (activityOrFlowId: string) => {
+      if (!respondentSubjectId) return;
+
+      setExpandedViewsLoading((prev) => ({ ...prev, [activityOrFlowId]: true }));
+
+      try {
+        const { data } = await getTargetSubjectsByRespondentApi({
+          activityOrFlowId,
+          subjectId: respondentSubjectId,
+        });
+        // Add or update the correspondent element in expandedViewsData
+        setExpandedViewsData((prev) => ({ ...prev, [activityOrFlowId]: data.result }));
+      } finally {
+        setExpandedViewsLoading((prev) => ({ ...prev, [activityOrFlowId]: false }));
+      }
+    },
+    [respondentSubjectId],
+  );
+
+  const handleRefetchActivities = useCallback(() => {
+    // Avoid fetching activities for respondent if respondent is a limited account
+    if (!appletId || !respondentSubject?.id || !respondentSubject.userId) return;
+
+    fetchActivities({ appletId, subjectId: respondentSubject.id });
+  }, [appletId, fetchActivities, respondentSubject]);
+
+  const handleRefetchAll = useCallback(() => {
+    handleRefetchActivities();
+
+    // Refresh target subject data for any expanded views
+    Object.entries(expandedViewsData).forEach(([id, viewData]) => {
+      if (viewData) handleRefetchExpandedView(id);
+    });
+  }, [handleRefetchActivities, expandedViewsData, handleRefetchExpandedView]);
+
+  const {
+    getActionsMenu,
+    onClickAssign,
+    onClickNavigateToData,
+    isLoading: isLoadingHook,
+    modals,
+  } = useAssignmentsTab({
+    appletId,
+    respondentSubject,
+    handleRefetch: handleRefetchAll,
+    dataTestId,
+  });
+
+  const handleClickNavigateToData = (
+    activityOrFlow: ParticipantActivityOrFlow,
+    targetSubject: RespondentDetails,
+  ) => {
+    if (!respondentSubject) return;
+
+    onClickNavigateToData(activityOrFlow, targetSubject.id);
+  };
+
+  const handleClickToggleExpandedView = useCallback(
+    (isExpanded: boolean, activityOrFlowId: string) => {
+      if (!respondentSubject) return;
+
+      if (isExpanded) {
+        handleRefetchExpandedView(activityOrFlowId);
+      } else {
+        // If expanded view is closed, remove data to free up memory and minimize refetches
+        // (after delay to account for transition)
+        setTimeout(() => {
+          setExpandedViewsData((prev) => ({ ...prev, [activityOrFlowId]: undefined }));
```

Comment:
Setting the entry to undefined will still keep it in the object, and later on `Object.entries.forEach` will still execute a re-fetch for it. Remove it instead
```suggestion
          setExpandedViewsData((prev) => {
            const { [activityOrFlowId]: _, ...rest } = prev;

            return rest;
          });
```